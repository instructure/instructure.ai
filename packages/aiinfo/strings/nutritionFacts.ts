// biome-ignore-all assist/source/useSortedKeys: Required order for Strings

import type { Strings } from "../types";

const strings: Strings = {
	en: {
		modalLabel: "This is a modal for AI facts",
		title: "Nutrition Facts",
		data: [
			{
				blockTitle: "Model & Data",
				segmentData: [
					{
						segmentTitle: "Base Model",
						description:
							"The foundational AI on which further training and customizations are built.",
					},
					{
						segmentTitle: "Trained with User Data",
						description:
							"Indicates the AI model has been given customer data in order to improve its results.",
					},
					{
						segmentTitle: "Data Shared with Model",
						description:
							"Indicates which training or operational content was given to the model.",
					},
				],
			},
			{
				blockTitle: "Privacy & Compliance",
				segmentData: [
					{
						segmentTitle: "Data Retention",
						description: "How long the model stores customer data.",
					},
					{
						segmentTitle: "Data Logging",
						description:
							"Recording the AI's performance for auditing, analysis, and improvement.",
					},
					{
						segmentTitle: "Regions Supported",
						description:
							"The locations where the AI model is officially available and supported.",
					},
					{
						segmentTitle: "PII",
						description:
							"Sensitive data that can be used to identify an individual.",
					},
				],
			},
			{
				blockTitle: "Outputs",
				segmentData: [
					{
						segmentTitle: "AI Settings Control",
						description:
							"The ability to turn the AI on or off within the product.",
					},
					{
						segmentTitle: "Human in the Loop",
						description:
							"Indicates if a human is involved in the AI's process or output.",
					},
					{
						segmentTitle: "Guardrails",
						description:
							"Preventative safety mechanisms or limitations built into the AI model.",
					},
					{
						segmentTitle: "Expected Risks",
						description: "Any risks the model may pose to the user.",
					},
					{
						segmentTitle: "Intended Outcomes",
						description:
							"The specific results the AI model is meant to achieve.",
					},
				],
			},
		],
		closeButtonText: "Close",
		closeIconButtonScreenReaderLabel: "Close",
		triggerText: "Nutrition Facts",
	},
};

export { strings };
