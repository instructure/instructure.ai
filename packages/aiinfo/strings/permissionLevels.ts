// biome-ignore-all assist/source/useSortedKeys: Required order for Strings

import type { DataPermissionLevelsStrings, Strings } from "../types";

const strings: Strings = {
	en: {
		closeButtonText: "Close",
		closeIconButtonScreenReaderLabel: "Close dialog",
		currentFeatureText: "Current feature",
		data: [
			{
				description:
					"We leverage anonymized aggregate data for detailed analytics to inform model development and product improvements. No AI models are used at this level.",
				level: "LEVEL 1",
				title: "Descriptive Analytics and Research",
			},
			{
				description:
					"We utilize off-the-shelf AI models and customer data as input to provide AI-powered features. No data is used for training this model.",
				level: "LEVEL 2",
				title: "AI-Powered Features Without Data Training",
			},
			{
				description:
					"We customize AI solutions tailored to the unique needs and resources of educational institutions. We use customer data to fine-tune data and train AI models that only serve your institution. Your institution's data only serves them through trained models.",
				level: "LEVEL 3",
				title: "AI Customization for Individual Institutions",
			},
			{
				description:
					"We established a consortium with educational institutions that shares anonymized data, best practices, and research findings. This fosters collaboration and accelerates the responsible development of AI in education. Specialized AI models are created for better outcomes in education, cost savings, and more.",
				level: "LEVEL 4",
				title: "Collaborative AI Consortium",
			},
		],
		modalLabel: "Data Permission Levels modal",
		title: "Data Permission Levels",
		triggerText: "Data Permission Levels",
	} as DataPermissionLevelsStrings,
};

export { strings };
