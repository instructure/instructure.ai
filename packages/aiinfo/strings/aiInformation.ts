// biome-ignore-all assist/source/useSortedKeys: Required order for Strings
import type { AiInformationStrings } from "../types";
import { strings as nutritionFactsStrings } from "./nutritionFacts";
import { strings as permissionLevelsStrings } from "./permissionLevels";

const strings: { en: AiInformationStrings } = {
	en: {
		title: "Features",
		data: {
			permissionLevelText: "Permission Level:",
			permissionLevelsModalTriggerText: permissionLevelsStrings.en.title,
			modelNameText:
				nutritionFactsStrings.en.data[0].segmentData[0].segmentTitle,
			nutritionFactsModalTriggerText: nutritionFactsStrings.en.title,
		},
		dataPermissionLevelsTitle: permissionLevelsStrings.en.title,
		dataPermissionLevelsCurrentFeatureText: "Current Feature:",
		dataPermissionLevelsCloseIconButtonScreenReaderLabel:
			permissionLevelsStrings.en.closeIconButtonScreenReaderLabel,
		dataPermissionLevelsCloseButtonText:
			permissionLevelsStrings.en.closeButtonText,
		dataPermissionLevelsModalLabel: permissionLevelsStrings.en.modalLabel,
		nutritionFactsModalLabel: nutritionFactsStrings.en.modalLabel,
		nutritionFactsTitle: nutritionFactsStrings.en.title,
		nutritionFactsCloseButtonText: nutritionFactsStrings.en.closeButtonText,
		nutritionFactsCloseIconButtonScreenReaderLabel:
			nutritionFactsStrings.en.closeIconButtonScreenReaderLabel,
	} as AiInformationStrings,
};

export { strings };
