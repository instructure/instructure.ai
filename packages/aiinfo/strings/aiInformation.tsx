// biome-ignore-all assist/source/useSortedKeys: Required order for Strings

import type { Renderable } from "@instructure/shared-types";
import type { AiInformationStrings } from "../types";
import { strings as nutritionFactsStrings } from "./nutritionFacts";
import { strings as permissionLevelsStrings } from "./permissionLevels";

const strings: { en: AiInformationStrings } = {
	en: {
		title: "Features",
		trigger: "<Button>AI Information</Button>" as Renderable,
		data: {
			permissionLevelText: "Permission Level:",
			permissionLevelsModalTriggerText: permissionLevelsStrings.en.title,
			modelNameText:
				nutritionFactsStrings.en.data[0].segmentData[0].segmentTitle,
			nutritionFactsModalTriggerText: nutritionFactsStrings.en.title,
		},
		dataPermissionLevelsTitle: permissionLevelsStrings.en.title,
		dataPermissionLevelsCurrentFeatureText: "Current Feature:",
		dataPermissionLevelsCloseIconButtonScreenReaderLabel:
			permissionLevelsStrings.en.closeIconButtonScreenReaderLabel,
		dataPermissionLevelsCloseButtonText:
			permissionLevelsStrings.en.closeButtonText,
		dataPermissionLevelsModalLabel: permissionLevelsStrings.en.modalLabel,
		nutritionFactsModalLabel: nutritionFactsStrings.en.modalLabel,
		nutritionFactsTitle: nutritionFactsStrings.en.title,
		nutritionFactsCloseButtonText: nutritionFactsStrings.en.closeButtonText,
		nutritionFactsCloseIconButtonScreenReaderLabel:
			nutritionFactsStrings.en.closeIconButtonScreenReaderLabel,
	} as AiInformationStrings,
};

export { strings };
