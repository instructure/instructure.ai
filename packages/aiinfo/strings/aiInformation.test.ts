import { describe, it, expect } from "vitest";
import { strings as aiInformationStrings } from "./aiInformation";
import { strings as nutritionFactsStrings } from "./nutritionFacts";
import { strings as permissionLevelsStrings } from "./permissionLevels";

describe("aiInformation strings", () => {
    it("exposes 'en' locale", () => {
        expect(aiInformationStrings).toHaveProperty("en");
    });

    it("has correct static literals", () => {
        const ai = aiInformationStrings.en;
        expect(ai.title).toBe("Features");
        expect(ai.trigger).toBe("<Button>AI Information</Button>");
        expect(String(ai.trigger)).toContain("<Button>");
    });

    it("maps data object fields correctly", () => {
        const ai = aiInformationStrings.en;
        const perm = permissionLevelsStrings.en;
        const nutrition = nutritionFactsStrings.en;
        expect(ai.data.permissionLevelText).toBe("Permission Level:");
        expect(ai.data.permissionLevelsModalTriggerText).toBe(perm.title);
        expect(ai.data.modelNameText).toBe(
            nutrition.data[0].segmentData[0].segmentTitle,
        );
        expect(ai.data.nutritionFactsModalTriggerText).toBe(nutrition.title);
    });

    it("maps permission level modal related fields", () => {
        const ai = aiInformationStrings.en;
        const perm = permissionLevelsStrings.en;
        expect(ai.dataPermissionLevelsTitle).toBe(perm.title);
        expect(ai.dataPermissionLevelsCurrentFeatureText).toBe("Current Feature:");
        expect(ai.dataPermissionLevelsCloseIconButtonScreenReaderLabel).toBe(
            perm.closeIconButtonScreenReaderLabel,
        );
        expect(ai.dataPermissionLevelsCloseButtonText).toBe(perm.closeButtonText);
        expect(ai.dataPermissionLevelsModalLabel).toBe(perm.modalLabel);
    });

    it("maps nutrition facts modal related fields", () => {
        const ai = aiInformationStrings.en;
        const nutrition = nutritionFactsStrings.en;
        expect(ai.nutritionFactsModalLabel).toBe(nutrition.modalLabel);
        expect(ai.nutritionFactsTitle).toBe(nutrition.title);
        expect(ai.nutritionFactsCloseButtonText).toBe(nutrition.closeButtonText);
        expect(ai.nutritionFactsCloseIconButtonScreenReaderLabel).toBe(
            nutrition.closeIconButtonScreenReaderLabel,
        );
    });

    it("all accessed fields are non-empty strings", () => {
        const ai = aiInformationStrings.en;
        const check = (val: unknown) =>
            expect(typeof val === "string" && val.length).toBeGreaterThan(0);

        check(ai.title);
        check(ai.trigger);
        check(ai.data.permissionLevelText);
        check(ai.data.permissionLevelsModalTriggerText);
        check(ai.data.modelNameText);
        check(ai.data.nutritionFactsModalTriggerText);
        check(ai.dataPermissionLevelsTitle);
        check(ai.dataPermissionLevelsCurrentFeatureText);
        check(ai.dataPermissionLevelsCloseIconButtonScreenReaderLabel);
        check(ai.dataPermissionLevelsCloseButtonText);
        check(ai.dataPermissionLevelsModalLabel);
        check(ai.nutritionFactsModalLabel);
        check(ai.nutritionFactsTitle);
        check(ai.nutritionFactsCloseButtonText);
        check(ai.nutritionFactsCloseIconButtonScreenReaderLabel);
    });

    it("has no undefined properties at top level of 'en'", () => {
        const ai = aiInformationStrings.en as Record<string, unknown>;
        for (const [k, v] of Object.entries(ai)) {
            expect(v).not.toBeUndefined();
        }
    });

    it("snapshot of english strings", () => {
        expect(aiInformationStrings.en).toMatchSnapshot();
    });
});