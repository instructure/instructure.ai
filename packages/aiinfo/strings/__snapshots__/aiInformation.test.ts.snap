// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`aiInformation strings > snapshot of english strings 1`] = `
{
  "data": {
    "modelNameText": "Base Model",
    "nutritionFactsModalTriggerText": "AI Nutrition Facts",
    "permissionLevelText": "Permission Level:",
    "permissionLevelsModalTriggerText": "Data Permission Levels",
  },
  "dataPermissionLevelsCloseButtonText": "Close",
  "dataPermissionLevelsCloseIconButtonScreenReaderLabel": "Close dialog",
  "dataPermissionLevelsCurrentFeatureText": "Current Feature:",
  "dataPermissionLevelsModalLabel": "Data Permission Levels modal",
  "dataPermissionLevelsTitle": "Data Permission Levels",
  "nutritionFactsCloseButtonText": "Close",
  "nutritionFactsCloseIconButtonScreenReaderLabel": "Close",
  "nutritionFactsModalLabel": "This is a modal for AI facts",
  "nutritionFactsTitle": "AI Nutrition Facts",
  "title": "Features",
  "trigger": "<Button>AI Information</Button>",
}
`;
